NUnit.Extension.AssemblyProgressReporter/TestSuiteMessage.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Xml;
+
+namespace NUnit.Extension.AssemblyProgressReporter {
+
+	internal enum TestSuiteType {
+		Assembly,
+		Namespace,
+		TestFixture,
+		ParameterizedTest
+	}
+
+	internal sealed class TestSuiteMessage : XmlMessage {
+
+		public TestSuiteType? Type { get; private set; }
+
+		public override MessageKind Kind => MessageKind.TestSuite;
+
+		public TestSuiteMessage( XmlReader reader ) : base( reader ) {
+			TestSuiteType tst;
+			Enum.TryParse( reader.GetAttribute( "type" ), out tst );

```

Comment:
Not checking result of TryParse, so you'll just get the first value.  Always check your try results